internal/internal.go

```diff
@@ -391,5 +391,32 @@ func parseTime(format string, timestamp string, location *time.Location) (time.T
 	case "stampnano":
 		format = time.StampNano
 	}
-	return time.ParseInLocation(format, timestamp, loc)
+
+	if !strings.Contains(format, "MST") {
+		return time.ParseInLocation(format, timestamp, loc)
+	}
```

Comment:
Not sure why this should be opt-in!? You currently get an unexpected and **wrong** time for all timestamps with a timezone-abbreviation as the time-offset to UTC is **always** zero no matter which timezone you have in your string... For example `2023-06-01 12:00:00 MST` would result in `2023-06-01T12:00:00+0000` which is clearly wron as it should be `2023-06-01T05:00:00+0000` or `2023-06-01T12:00:00-0700` . Even worse printing the timestamp will display the original string as the zone abbreviation is kept but the offset is zero. I can make this an opt-in but I really don't see why someone wants this buggy behavior...

Reply:
Correct, this is a bug fix, not a new feature..

Reply:
Then I need to request two additions:

1) An agent-level option that sets this new behavior. Unlike other options you can set the default to the new behavior, but the user needs to have a way to opt-out.
2) Documentation that shows a user how to set the zone info.

While I understand you see the change as fixing wrong behavior, it does not fully correct the behavior for all our users. You said below a user should not use MST to mean Malaysian standard time, however that is an entirely valid use case and responding to a user to "not do that" is entirely unacceptable. Documentation will help resolve this.

Users who grab the new version of telegraf with this change will see timestamps that are affected by this, suddenly change. If a user comes to you and says you broke all my alerting or reporting because of this, then what? The option to opt-out will resolve this.

Reply:
These users need to set a location or timezone, as they are probably already doing since otherwise they would have wrong timestamps..

For CSV parsing this is done with `csv_timezone`, other parsers have similar option.

I agree to have documentation on how to set this advanced timezone info..

Reply:
@powersj I agree on the documentation point (item 2) but I strongly disagree with the option. The current situation is that everyone using abbreviations **now get the wrong time** and this also holds for MST meaning Malaysian! Even worse, currently we accept arbitrary timezone abbreviations, even those that do not exists e.g. `XYZ`, without error and **completely ignoring those** and return a time that is in UTC (offset zero). So your MST time is wrong and always as if you provided `UTC`. Relying on buggy behavior is bad and we will carry this flag till doomsday... :-(

Reply:
I am not sure why you would strongly disagree with an opt-out feature ;) It is entirely optional, but gives users who get affected and need time to update a way out.

I do not disagree that the current situation is wrong behavior, but this change is breaking behavior. If we had broken this even 4 or 5 minor releases ago and are now fixing this, then this would not be needed.